nat-lab/tests/test_events.py

```diff
@@ -679,3 +679,289 @@ async def test_event_content_meshnet_node_upgrade_direct(
+            marks=pytest.mark.windows,
+        ),
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.MAC_VM,
+                adapter_type=AdapterType.BoringTun,
+                connection_tracker_config=generate_connection_tracker_config(
+                    ConnectionTag.MAC_VM,
+                    derp_1_limits=ConnectionLimits(1, 1),
+                ),
+                features=TelioFeatures(
+                    no_link_detection=NoLinkDetection(rtt_seconds=30)
+                ),
+            ),
+            marks=pytest.mark.mac,
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "beta_setup_params",
+    [
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_2,
+                connection_tracker_config=generate_connection_tracker_config(
+                    ConnectionTag.DOCKER_CONE_CLIENT_2,
+                    derp_1_limits=ConnectionLimits(1, 1),
+                ),
+            )
+        )
+    ],
+)
+async def test_event_link_state_with_configured_rtt(
+    alpha_setup_params: SetupParameters, beta_setup_params: SetupParameters
+) -> None:
+    async with AsyncExitStack() as exit_stack:
+        env = await setup_mesh_nodes(
+            exit_stack, [alpha_setup_params, beta_setup_params]
+        )
+        alpha, beta = env.nodes
+        client_alpha, client_beta = env.clients
+        connection_alpha, connection_beta = [
+            conn.connection for conn in env.connections
+        ]
+
+        async with Ping(connection_alpha, beta.ip_addresses[0]).run() as ping:
+            await testing.wait_long(ping.wait_for_next_ping())
+        async with Ping(connection_beta, alpha.ip_addresses[0]).run() as ping:
+            await testing.wait_long(ping.wait_for_next_ping())
+
+        beta_state = client_alpha.get_node_state(beta.public_key)
+        assert beta_state is not None and beta_state.link_state == LinkState.Up
+        assert beta_state == PeerInfo(
+            identifier=beta.id,
+            public_key=beta.public_key,
+            state=State.Connected,
+            link_state=LinkState.Up,
+            is_exit=False,
+            is_vpn=False,
+            ip_addresses=beta.ip_addresses,
+            allowed_ips=env.api.get_allowed_ip_list(beta.ip_addresses),
+            nickname=None,
+            endpoint=None,
+            hostname=beta.name + ".nord",
+            allow_incoming_connections=True,
+            allow_peer_send_files=True,
+            path=PathType.Relay,
+        )
+
+        # Kill beta node
+        await client_beta.stop_device()
+
+        # As we have configured the rtt to 30 reconds we should not receive any event
+        # in 10 (WG_KEEPALIVE) + 30 (configured_rtt) seconds
+        with pytest.raises(asyncio.TimeoutError):
+            await client_alpha.wait_for_link_state_event(
+                beta.public_key, [LinkState.Down], timeout=39
+            )
+
```

Comment:
Just curious, can't we make the configured rtt to like 1 second and expect the event in ~11-12 seconds? Should yield quicker test time. It doesn't matter if the test runner is parallelized that much though.